app/models/site.rb

```diff
@@ -50,7 +50,7 @@ def select_proxy(older_than=-1)
         Proxy::NoProxy
       elsif proxy_ts > threshold_ts
         # The proxy we found was too recently used.
-        Proxy::NoColdProxy.new(proxy_ts - threshold_ts)
+        Proxy::NotReady.new(proxy_ts, threshold_ts, proxy_id)
```

Comment:
Shouldn't we touch the proxy after it's reserved? That way we minimize the chance of it being reserved by multiple clients.